reference/docker/sort.go

```diff
@@ -56,6 +56,43 @@ func Sort(references []string) []string {
 	return append(refs, bad...)
 }
 
+// StrictSort sorts string references preferring higher information references
+// The precedence is as follows:
+// 1. Name + Tag + Digest
+// 2. Name + Tag
+// 3. Name + Digest
+// 4. Name
+// 5. Digest
+// 6. Parse error and inconsistent ref
+// Unlike Sort, it does not modify the items of `references`
+func StrictSort(references []string) []string {
```

Comment:
The best way to do this, would be to remove `Sort` and rename `StrictSort` to `Sort`, but this would be an incompatible change for the `Sort` function, so I kept `Sort` and marked it as deprecated